### DISCUSSION TITLE
How to make alias call multiple functions (tap-next and cmd-button)

### DISCUSSION THREAD

**Discussion by michaelmoreno** [kmonad/kmonad#590] [General]
I have an alias that switches to a layer for navigation (rebind JIKL to arrow keys)
```lisp
(defalias
  nav (tap-next (layer-switch navigation) lmet))
```
Sometimes I forget when I've switched to this layer,  so I'd like to also turn on my keyboard backlight on activation with this command
```lisp
(cmd-button "echo 1 | sudo tee /sys/class/leds/asus::kbd_backlight/brightness")
```
However I'm not sure how to make the `nav` alias call both of these functions at the same time. I've tried
```lisp
(defalias
  nav (tap-next (layer-switch navigation) lmet) (cmd-button "echo 1 | sudo tee /sys/class/leds/asus::kbd_backlight/brightness"))
```
and 
```lisp
(defalias
  nav (tap-next (layer-switch navigation (cmd-button "echo 1 | sudo tee /sys/class/leds/asus::kbd_backlight/brightness")) lmet))
```
which did not work.

I've googled around for running multiple functions with an alias in KMonad and did not find anything.

**Comment by MuhammedZakir** [comment]
`tap-macro`, `around` and/or `around-next` are used for these.

E.g.
```clojure
(defalias
  led-switch-nav #((cmd-button "command to set led") (layer-switch navigation))
  nav (tap-next @led-switch-nav lmet)
)
```

Edit: fixed: forgot `@` before alias.
